xdsl/transforms/convert_snrt_to_riscv.py

```diff
@@ -198,8 +201,95 @@ def match_and_rewrite(
         )
 
 
+class LowerDMAStart2DBase(RewritePattern, ABC):
+    any_reg = riscv.IntRegisterType.unallocated()
+
+    def gen_insns(
+        self,
+        dst_low: SSAValue | Operation,
+        dst_high: SSAValue | Operation,
+        src_low: SSAValue | Operation,
+        src_high: SSAValue | Operation,
+        size: SSAValue | Operation,
+        dst_stride: SSAValue | Operation,
+        src_stride: SSAValue | Operation,
+        repeat: SSAValue | Operation,
+    ) -> tuple[Sequence[Operation], Sequence[SSAValue]]:
+        # dmsrc a0, a1
+        # dmdst a0, a1
+        # dmstr a5, a6
+        # dmrep a7
+        # dmcpyi a0, a4, 0b10
+        return [
+            riscv_snitch.DMSourceOp(src_low, src_high),
+            riscv_snitch.DMDestinationOp(dst_low, dst_high),
+            riscv_snitch.DMStrideOp(src_stride, dst_stride),
+            riscv_snitch.DMRepOp(repeat),
+            cpy_op := riscv_snitch.DMCopyImmOp(size, 0b10),
+            tx_id := builtin.UnrealizedConversionCastOp.get([cpy_op], [builtin.i32]),
+        ], tx_id.results
+
+    def cast_i32(self, input_val: SSAValue):
+        return builtin.UnrealizedConversionCastOp.get([input_val], [self.any_reg])
+
+    def cast_i64(self, input_val: SSAValue):
```

Comment:
Properly casting 64-bit integers to 32-bit RISC-V registers is not implemented right now. Future work!